bundle/config/mutator/override_compute.go

```diff
@@ -38,24 +39,34 @@ func overrideJobCompute(j *resources.Job, compute string) {
 }
 
 func (m *overrideCompute) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
-	if b.Config.Bundle.Mode != config.Development {
+	var diags diag.Diagnostics
+
+	if b.Config.Bundle.Mode == config.Production {
 		if b.Config.Bundle.ClusterId != "" {
-			return diag.Errorf("cannot override compute for an target that does not use 'mode: development'")
+			// Overriding compute via a command-line flag for production works, but is not recommended.
+			diags = diags.Extend(diag.Warningf("Overriding compute for a target that uses 'mode: production' is not recommended"))
```

Comment:
This warning is difficult to map to either `--cluster-id` or the `bundle.cluster_id` setting without additional information like the config path or location information. Alternatively, the message could be updated to not talk about compute but cluster.

It would also be good to include a "detail" section to explain why it is not recommended.